docs/dev/core/new-metric-type.md

```diff
@@ -0,0 +1,212 @@
+# Adding a new metric type
+
+Data in Glean is stored in so called metrics.
+You can find the full list of implemented metric types [in the user overview](/user/metrics/index.md).
+
+Adding a new metric type involves defining the metric type's API, its persisted and in-memory storage as well as its serialization into the ping payload.
+
+## The metric type's API
+
+A metric type implementation is defined in its own file under `glean-core/src/metrics/`, e.g. `glean-core/src/metrics/counter.rs` for a [Counter](/user/metrics/counter.md).
+
+Start by defining a structure:
```

Comment:
... to hold the metric's metadata.

(Or something, to make it clear that this structure is holding the metric definition, not the data for each recording of the metric)...